core/src/main/scala/pureconfig/package.scala

```diff
@@ -143,9 +166,13 @@ package object pureconfig {
    */
   @throws[ConfigReaderException[_]]
   def loadConfigOrThrow[Config](path: Path)(implicit conv: ConfigConvert[Config], ct: ClassTag[Config]): Config = {
-    ConfigFactory.invalidateCaches()
-    val config = ConfigFactory.load(ConfigFactory.parseFile(path.toFile))
-    getResultOrThrow[Config](loadConfig[Config](config)(conv))
+    val errorOrConfig =
+      for {
+        _ <- invalidateCaches().right
+        rawConfig <- loadFile(path).right
+        config <- loadConfig[Config](rawConfig)(conv).right
+      } yield config
+    getResultOrThrow[Config](errorOrConfig)
```

Comment:
> I'm not sure why the first set the position of the error to None while the second to ConfigValueLocation(rawConfig.root()) but this is preventing me from reusing the first in the second. @jcazevedo do you know why the two methods are different for the error location?

It looks like a mistake. I think the `location` on `loadConfig` should be obtained via `ConfigValueLocation(rawConfig.root())` as well.